File: WS_EUREKABANK_SOAPJAVA_GR04/02.CLIESC/CLIESC_EurekaBank_SOAP_GR04/src/main/java/ec/edu/monster/controller/LoginController.java
package ec.edu.monster.controller;

import ec.edu.monster.service.EurekaApi;
import ec.edu.monster.service.SoapEurekaApi;
import ec.edu.monster.view.HomeView;
import ec.edu.monster.view.LoginView;

public class LoginController {
  private final LoginView view;
  private final EurekaApi api;

  public LoginController(LoginView view) {
    this.view = view;
    this.api  = new SoapEurekaApi();
    bind();
  }

  private void bind() {
    this.view.onLogin((u,p) -> {
      try {
        boolean ok = api.validarIngreso(u, p);
        if (!ok) { view.showError("Credenciales inválidas."); return; }
        Session.usuario = u;
        view.dispose();
        new HomeController(new HomeView(), api).show();
      } catch (Exception ex) {
        view.showError("Error conectando al servicio: " + ex.getMessage());
      }
    });
  }

  /** Mostrar la pantalla de login */
  public void show() {
    view.setVisible(true);
  }
}

File: WS_EUREKABANK_RESTFULLNOTNET_GR04/02.CLIESC/CLIESC_EurekaBank_REST_GR04/CLIESC_EurekaBank_REST_GR04/ec.edu.monster.controllers/LoginController.cs
using System.Threading.Tasks;
using System.Windows;
using ec.edu.monster.services;
using ec.edu.monster.views;

namespace ec.edu.monster.controllers
{
    public class LoginController
    {
        private readonly LoginWindow _view;
        private readonly IEurekaApi _api;

        public static string UsuarioActual { get; private set; }

        public LoginController(LoginWindow view, IEurekaApi api)
        {
            _view = view;
            _api = api;

            _view.LoginClicked += async (u, p) => await DoLogin(u, p);
        }

        private async Task DoLogin(string user, string pass)
        {
            if (string.IsNullOrWhiteSpace(user) || string.IsNullOrWhiteSpace(pass))
            {
                _view.ShowError("Ingresa usuario y contraseña.");
                return;
            }

            // Bypass para demo sin servidor: acepta MONSTER / MONSTER9
            if (string.Equals(user, "MONSTER", StringComparison.OrdinalIgnoreCase) &&
                string.Equals(pass, "MONSTER9", StringComparison.OrdinalIgnoreCase))
            {
                UsuarioActual = user.ToUpperInvariant();
                var home = new HomeWindow();
                var ctrl = new HomeController(home, _api);
                System.Windows.Application.Current.MainWindow = home;
                _view.Hide();
                home.Show();
                _view.Close();
                return;
            }

            _view.SetBusy(true);
            try
            {
                var ok = await _api.ValidarIngreso(user, pass);
                if (!ok)
                {
                    _view.ShowError("Credenciales inválidas.");
                    return;
                }

                UsuarioActual = user.ToUpperInvariant();
                var home = new HomeWindow();
                var ctrl = new HomeController(home, _api);
                _view.Hide();
                home.Show();
                _view.Close();
            }
            catch (System.Exception ex)
            {
                MessageBox.Show("Error conectando al servicio: " + ex.Message,
                    "Login", MessageBoxButton.OK, MessageBoxImage.Error);
            }
            finally
            {
                _view.SetBusy(false);
            }
        }
    }
}
